### ISSUE TITLE
Port request for ...

### ISSUE THREAD

**Issue by dserrato** [backdrop-ops/contrib#456]
**Name of the module, theme, or layout**
Relation

**Link to the drupal.org module, theme, or layout**
https://www.drupal.org/project/relation

**(Optional) Link to an issue in the drupal.org queue asking the Drupal community if anyone is working on a Backdrop port of this project**
I did not find any information in the issue queue, but that's probably because Drupal is moving to replace Relation with Group and Dynamic Entity Reference for D8 and forward.

**(Optional) Bounty**
At this time, I don't have any idea if we could provide bounty funds

**Comment by jenlampton** [comment]
What is the difference between relation / references / entity reference modules in Drupal? And, have you looked into the reference module for backdrop and does it do what you need? If so, I would be interested in working on an upgrade path from relation -> reference.

**Comment by stpaultim** [comment]
@jenlampton - I once used the relations module in Drupal 7, it's been a while. But, I think the selling point was that it creates a NEW entity links two other entity types. It was my understanding at the time, that this gave us more flexibility in creating connections from either end. For example, if you have a class content type and and instructor content type. You can use the relations module to connect a class to a teacher OR a teacher to a class. 

I don't think that is possible with the references module. I believe that with references, you can connect classes to teachers or teachers to classes, but you can't go both ways.

**Comment by dserrato** [comment]
That is correct. My example is that we use relation is so that we can connect and create relationships between people, and extend that relation to a third person. So, we have a teacher that has a mentor/coach. The coach will mentor several different teachers, independently. The teachers don't have any relation to one another. To extend that, we then have the Lead Coach that oversees the Coaches, and that translates down to the teachers. Lots and lots of relationships that are transitory. 

Like family trees:
Grandparent
-> Parent
->-> Child A
->-> Child B

Because Child A is a related to the Parent, there is then a transitory connection to the Grandparent. These are all relations and can be queried, viewed, and connected in all sorts of ways. 

Using entity references, you are just connecting maybe one node to another. 

I have two projects in D7 that rely heavily on this and I was looking at the possibility of moving them to Backdrop given that Drupal is moving a different direction in this kind of area. I figured I would look into what it might take to delve into a conversion, but I'm going to assume this one might be pretty big to move. 

Thank you for your consideration and beginning a conversation around this. Have a great weekend.
Dan

**Comment by jenlampton** [comment]
Thank you, these use-cases are very clearly explained. I agree that this needs to remain a separate project. 

I wonder if the fields on the relation entity should be converted to the new reference fields? Perhaps not in the initial port, but at some point in the future. :)

**Comment by stpaultim** [comment]
Once again, I'm running into a situation in which I think this module would be really useful. I can probably make do with references, but the relation module would be nice.

**Comment by robertgarrigos** [comment]
I use a lot the references module and would use this relation module definitely.
